### PR TITLE
[SOURCE] Payhere Subscriber hits “Submit” button #1086

### PR DESCRIPTION
**Pull request by lcaresia**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/payhere/payhere.app.mjs [hunk 1]
```diff
@@ -1,11 +1,44 @@
...
+        method: "post",
+        data,
+      });
+
+      return response.data;
+    },
+    async removeWebhook(webhookId) {
+      return this._makeRequest(`hooks/${webhookId}`, {
+        method: "delete",
+      });
```
Comment (jcortes) [inline components/payhere/payhere.app.mjs:41]: Optional:
To simplify a bit more:
```suggestion
    async _makeRequest({
      $ = this, path, ...args
    } = {}) {
      return axios($, {
        url: `${this._apiUrl()}${path}`,
        headers: {
          Authorization: `Bearer ${this._apiKey()}`,
        },
        ...args,
      });
    },
    async createWebhook(args = {}) {
      return this._makeRequest({
        path: "/hooks",
        method: "post",
        ...args,
      });
    },
    async removeWebhook({
      webhookId, ...args
    } = {}) {
      return this._makeRequest({
        path: `/hooks/${webhookId}`,
        method: "delete",
        ...args,
      });
```

**Review by jcortes — CHANGES_REQUESTED** [review]
Hi @lcaresia I've added just one minor request change and an observation. Thanks!

**Review by jcortes — COMMENTED** [review]
Hi @lcaresia it looks good to me now! Ready for QA!

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**